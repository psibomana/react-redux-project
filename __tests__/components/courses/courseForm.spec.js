import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { createMemoryHistory } from 'history'
import CourseForm from '../../../src/components/course/courseForm'

describe('Course Form Test', () => {
  let wrapper, course, mockOnChange, mockOnSave, mockOnDelete, history;

  beforeEach(() => {
    course = {
        id: "clean-code",
        title: "Clean Code: Writing Code for Humans",
        watchHref: "http://www.pluralsight.com/courses/writing-clean-code-humans",
        authorId: "cory-house",
        length: "3:10",
        category: "Software Practices"
      }

    mockOnChange = jest.fn();
    mockOnSave = jest.fn();
    mockOnDelete = jest.fn();
    history = createMemoryHistory('/authors')

    wrapper = shallow(
      <CourseForm
        course={course}
        errors={{}}
        allAuthors={[]}
        onChange={mockOnChange}
        onSave={mockOnSave}
        onDelete={mockOnDelete}
        saving={false}
        history={history}
        />
    )
  })

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('should call onChange handler from title', () => {

    const event = {
      target: { value: 'updated-title' }
    };

    wrapper.find('#title').simulate('change', event);
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });

  it('should call onChange handler from author input', () => {

    const event = {
      target: { value: 'updated-author' }
    };

    wrapper.find('#clean-code').simulate('change', event);
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });

  it('should call onChange handler from category', () => {

    const event = {
      target: { value: 'updated-category' }
    };

    wrapper.find('#category').simulate('change', event);
    expect(mockOnChange.mock.calls.length).toEqual(1);
  });
})
