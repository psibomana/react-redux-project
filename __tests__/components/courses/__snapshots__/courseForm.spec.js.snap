// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Form Test should render as expected 1`] = `
<form>
  <h1>
    Manage Course
  </h1>
  <TextInput
    id="title"
    label="Title"
    name="title"
    onChange={[Function]}
    value="Clean Code: Writing Code for Humans"
  />
  <SelectInput
    defaultOption="Select Author"
    id="clean-code"
    label="Author"
    name="authorId"
    onChange={[Function]}
    options={Array []}
    value="cory-house"
  />
  <TextInput
    id="category"
    label="Category"
    name="category"
    onChange={[Function]}
    value="Software Practices"
  />
  <TextInput
    label="Length"
    name="length"
    onChange={[Function]}
    value="3:10"
  />
  <input
    className="btn btn-primary"
    disabled={false}
    id="saveBtn"
    onClick={[Function]}
    type="button"
    value="Save"
  />
  <input
    className="btn btn-danger"
    id="deleteBtn"
    onClick={[Function]}
    type="button"
    value="Delete"
  />
</form>
`;
